src/main/java/com/google/devtools/build/bfg/bfg.proto

```diff
@@ -0,0 +1,19 @@
+syntax = "proto2";
+
+package com.google.devtools.build.bfg;
+
+option java_package = "protos.com.google.devtools.build.bfg";
+
+// ParserOutput is used to communicate between language-specific parsers and the language-agnostic
+// BFG core.
+message ParserOutput {
+    // The class dependency graph.
+    // In the Java case, we have (u, v) if 'u' mentions [1] 'v'.
+    // [1] - By mention, we mean either imports or references in code, either as a fully qualified
+    //       name, or an implied same-package name.
+    map<string, Strings> class_to_class = 1;
+}
+
+message Strings {
```

Comment:
like above, can you make this some variant of `dependency` like `deps` or what have you?

Reply:
I'm going to reuse this for other graphs (class_to_file, file_to_rule_kind) so "dependency" doesn't fit. 
OTOH defining extra messages just for the name sounds wasteful.

Reply:
I am fine accepting `class_to_class`, but `Strings` (and `s`) is... sketchy. It's overly broad.

> OTOH defining extra messages just for the name sounds wasteful.

Understandable complaint. However, using overly broad or single letter variable names in Java is inappropriate and non-idiomatic. Perhaps it's not an issue when looking at the proto in isolation, but it's hard to argue the below is proper java code
```java
private static HashMap<String, Bfg.Strings> writeGraphToProtoMap; 
// you defined your own special Strings class. Why???
...

adj.addAllS(graph.adjacentNodes(u));
// what is S???
```

> I'm going to reuse this for other graphs (class_to_file, file_to_rule_kind) 

there has to be some unifying quality about each of those, beyond `Strings`. Perhaps node.

As a side note, if we move to scala, i will be l

Reply:
How about `repeated string element` instead of `repeated string s`?

Reply:
That works